### PR TITLE
fix(affiliate): Use DB transaction for username generation; add more stats for update…

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced transaction management for username generation and affiliate info updates.
	- Introduced timing metrics for improved performance tracking during tasks.

- **Bug Fixes**
	- Improved error handling for username collisions and insertion failures.

- **Tests**
	- Updated test cases for username generation to reflect new expected outputs and improve clarity.
	- Modified tests for subaccount username generation to align with the latest username changes.

- **Chores**
	- Removed and updated adjectives and nouns in the respective JSON files, impacting the generation process.

### CODE REVIEW

File: indexer/services/roundtable/src/tasks/subaccount-username-generator.ts [hunk 1]
```diff
@@ -2,85 +2,105 @@ import { logger, stats } from '@dydxprotocol-indexer/base';
...
+      message: 'Generated usernames',
+      batchSize: subaccountZerosWithoutUsername.length,
+      successCount,
+      addressSample: subaccountAddresses.slice(0, 10),
+      duration: Date.now() - taskStart,
+    });
+  } catch (error) {
+    await Transaction.rollback(txId);
+    logger.error({
+      at: 'update-wallet-total-volume#runTask',
```
Comment (vincentwschau) [inline indexer/services/roundtable/src/tasks/subaccount-username-generator.ts:96]: nit: Update name to `subaccount-username-generator`.

**Review by vincentwschau — APPROVED** [review]

File: indexer/services/roundtable/src/helpers/adjectives.json [hunk 2]
```diff
@@ -149,10 +149,8 @@
   "hip",
   "holy",
   "honest",
-  "hot",
```
Comment (teddyding) [inline indexer/services/roundtable/src/helpers/adjectives.json:152]: Refactored both word lists and spot checked 34 [proposed whitelist addresses ](https://www.mintscan.io/dydx/proposals/183)

**Review by vincentwschau — APPROVED** [review]

**Comment by teddyding** [comment]
@Mergifyio release/indexer/v7.x

**Comment by teddyding** [comment]
@Mergifyio backport release/indexer/v7.x

**Pull request merged**